airbyte-cdk/java/airbyte-cdk/s3-destinations/src/main/kotlin/io/airbyte/cdk/integrations/destination/s3/S3ConsumerFactory.kt

```diff
@@ -221,25 +220,52 @@ class S3ConsumerFactory {
                 memoryRatio
             }
 
+        // This needs to be called before the creation of the flush function because it updates
+        // writeConfigs!
+        val onStartFunction = onStartFunction(storageOps, writeConfigs)
+
+        val streamDescriptorToWriteConfig =
+            writeConfigs.associateBy {
+                StreamDescriptor().withNamespace(it.namespace).withName(it.streamName)
+            }
+        val flushFunction =
+            if (featureFlags.useFileTransfer()) {
+                FileTransferDestinationFlushFunction(
```

Comment:
What determines whether the feature flag is set? The fact that the source is flagged as a file source? Explicit opt-in at the sync level?

Reply:
Fair question.

Basically, we need the source configuration to have a specific parameter enabled (I don't know the details of the parameter) AND the destination needs to have `supportsFileTransfer` set to `true` in its `metadata.yaml`. If those 2 conditions are true, then the 2 variables are set accordingly, a common volume is mounted on both containers, and it's expected that all records are file-based instead of record-based.
If the source config has the parameter set to `true` and the destination doesn't support file transfer, the platform will throw an exception